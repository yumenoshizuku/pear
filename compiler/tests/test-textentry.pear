Main()(
window.Window(),
window.Position("center"),
window.Title("Text Entry"),
window.BorderWidth(10),
table.Grid(),
table.Hom("row"),
table.Hom("column"),
window.Contain(table),
label1.Label("Name"),
label2.Label("Age"),
label3.Label("Occupation"),
table.AttachAt(label1,0,0,1,1),
table.AttachAt(label2,0,1,1,1),
table.AttachAt(label3,0,2,1,1),
entry1.TextEntry(),
entry2.TextEntry(),
entry3.TextEntry(),
table.AttachAt(entry1,1,0,1,1),
table.AttachAt(entry2,1,1,1,1),
table.AttachAt(entry3,1,2,1,1),
window.Showall(),
)
