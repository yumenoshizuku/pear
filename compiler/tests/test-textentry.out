#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>
#include <glib.h>
#include <glib/gstdio.h>


int main(int argc, char** argv)
{
	GtkWidget* window;
	GtkWidget* table;
	GtkWidget* label1;
	GtkWidget* label2;
	GtkWidget* label3;
	GtkWidget* entry1;
	GtkWidget* entry2;
	GtkWidget* entry3;
	
	gtk_init(&argc, &argv);
	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
	g_signal_connect(window, "destroy", G_CALLBACK(gtk_main_quit), NULL);
	gtk_window_set_position(GTK_WINDOW(window), GTK_WIN_POS_CENTER);
	gtk_window_set_title(GTK_WINDOW(window), "Text Entry");
	gtk_container_set_border_width(GTK_CONTAINER(window), 10);
	table = gtk_grid_new();
	gtk_grid_set_row_homogeneous(GTK_GRID(table), TRUE);
	gtk_grid_set_column_homogeneous(GTK_GRID(table), TRUE);
	gtk_container_add(GTK_CONTAINER(window), table);
	label1 = gtk_label_new ("Name");
	label2 = gtk_label_new ("Age");
	label3 = gtk_label_new ("Occupation");
	gtk_grid_attach(GTK_GRID(table), label1, 0, 0, 1, 1);
	gtk_grid_attach(GTK_GRID(table), label2, 0, 1, 1, 1);
	gtk_grid_attach(GTK_GRID(table), label3, 0, 2, 1, 1);
	entry1 = gtk_entry_new();
	entry2 = gtk_entry_new();
	entry3 = gtk_entry_new();
	gtk_grid_attach(GTK_GRID(table), entry1, 1, 0, 1, 1);
	gtk_grid_attach(GTK_GRID(table), entry2, 1, 1, 1, 1);
	gtk_grid_attach(GTK_GRID(table), entry3, 1, 2, 1, 1);
	gtk_widget_show_all(window);
	gtk_main();
	return 0;
}

